File: 이진욱/2주차/10815.py
def binarySearch(arr, num):
    start = 0
    end = len(arr) - 1
    while start <= end:
        mid = (start + end) //2
        #찾고자 하는 값 보다 클 경우
        if num < arr[mid]:
            end = mid - 1
        #찾고자 하는 값 보다 작을 경우
        elif num > arr[mid]:
            start = mid + 1
        else:
            return 1
    return 0

n = input()
hold = list(map(int, input().split()))
hold.sort()
m = input()
find = list(map(int, input().split()))
answer = []

for i in find:
    answer.append(binarySearch(hold, i))

for i in answer:
    print(i, end=" ")

File: 한지석/2주차/10815.swift
//
//  10815.swift
//  2주차
//
//  Created by 한지석 on 2022/09/26.
//

import Foundation

func binarySearch(array: [Int], num: Int) {
    var start = 0
    var end = (array.count - 1)
    
    while start <= end {
        let mid = (start + end) / 2
        
        if array[mid] == num {
            print("1", terminator: " ")
            return
        }
        if array[mid] > num {
            end = mid - 1
        } else {
            start = mid + 1
        }
    }
    print("0", terminator: " ")
}

let N = Int(readLine()!)!

var input1 = readLine()!.components(separatedBy: " ")
var card = input1.map { Int(String($0))! }
card.sort()

let M = Int(readLine()!)!
var input2 = readLine()!.components(separatedBy: " ")
var inputCard = input2.map { Int(String($0))! }

for index in 0..<M {
    binarySearch(array: card, num: inputCard[index])
}
